reporter.go

```diff
@@ -356,15 +354,15 @@ func (r *Reporter) Compare(before, after *fspb.Walk) (*Report, error) {
 	walkedAfter := map[string]*fspb.File{}
 	if before != nil {
 		for _, fbOrig := range before.File {
-			fb := *fbOrig
+			fb := proto.Clone(fbOrig).(*fspb.File)
 			fb.Path = NormalizePath(fb.Path, fb.Info.IsDir)
-			walkedBefore[fb.Path] = &fb
+			walkedBefore[fb.Path] = fb
```

Comment:
happy to remove this change and the one below, and instead consider that separately.

Reply:
I like these changes, it's how we should be cloning protos to begin with.